### ISSUE TITLE
Extract namespace for `<Translation>` render prop

### ISSUE THREAD

**Issue by ggalletty** [i18next/i18next-parser#702]
## 🚀 Feature Proposal

Add coverage for `<Translation>` render prop component and support parsing default namespace from its props. 

## Motivation

Currently the parser doesn't take into account `<Translation>` namespace for children expressions within the render prop.

## Example

With this feature we would extract namespace defined in `ns` prop.

Given

```
<Translation ns="foo">{(t) => t("some_key")}</Translation>
```

The JS lexer would be able to assign the proper namespace to extracted keys
```
{ key: "some_key", namespace: "foo" }
```

**Comment by ggalletty** [comment]
hi @karellm! bumping this in hopes we can review it and hopefully merge related change request 🤞

i believe this would be a good and cheap addition since translation render prop is not supported at all

**Comment by karellm** [comment]
This is published as `7.3.0`
